homeassistant/components/sonos/diagnostics.py

```diff
@@ -91,9 +91,13 @@ async def async_generate_media_info(
         payload[attrib] = getattr(speaker.media, attrib)
 
     def poll_current_track_info():
-        return speaker.soco.avTransport.GetPositionInfo(
-            [("InstanceID", 0), ("Channel", "Master")]
-        )
+        try:
+            return speaker.soco.avTransport.GetPositionInfo(
+                [("InstanceID", 0), ("Channel", "Master")],
+                timeout=3,
+            )
+        except OSError:
+            return "Error retrieving"
```

Comment:
```suggestion
        except OSError as ex:
            return f"Error retrieving: {ex}"
```

Should we put the exception message in the diagnostics so we know why its missing?

Reply:
Sure.